src/Nethermind/Nethermind.Core.Test/Encoding/BlockDecoderTests.cs

```diff
@@ -86,6 +87,18 @@ public BlockDecoderTests()
                 .WithBlobGasUsed(ulong.MaxValue)
                 .WithExcessBlobGas(ulong.MaxValue)
                 .WithMixHash(Keccak.EmptyTreeHash)
+                .TestObject,
+            Build.A.Block.WithNumber(1)
+                .WithBaseFeePerGas(1)
+                .WithTransactions(transactions)
+                .WithUncles(uncles)
+                .WithWithdrawals(8)
+                .WithBlobGasUsed(ulong.MaxValue)
+                .WithExcessBlobGas(ulong.MaxValue)
+                .WithMixHash(Keccak.EmptyTreeHash)
+                // an empty Deposit array
+                .WithDeposits(0)
+                .WithValidatorExits(new[] { new ValidatorExit(TestItem.AddressA, new byte[48]) })
```

Comment:
add tests with different edge cases for deposits/exits:
null
0
1
more than 1